Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -0,0 +1,264 @@
+import Foundation
+import KsApi
+import PassKit
+import Prelude
+import ReactiveSwift
+
+public protocol ConfirmDetailsViewModelInputs {
+  func configure(with data: PledgeViewData)
+  func pledgeAmountViewControllerDidUpdate(with data: PledgeAmountData)
+  func shippingRuleSelected(_ shippingRule: ShippingRule)
+  func viewDidLoad()
+}
+
+public protocol ConfirmDetailsViewModelOutputs {
+  var configureLocalPickupViewWithData: Signal<PledgeLocalPickupViewData, Never> { get }
+  var configurePledgeAmountViewWithData: Signal<PledgeAmountViewConfigData, Never> { get }
+  var configureShippingLocationViewWithData: Signal<PledgeShippingLocationViewData, Never> { get }
+  var configureShippingSummaryViewWithData: Signal<PledgeShippingSummaryViewData, Never> { get }
+  var localPickupViewHidden: Signal<Bool, Never> { get }
+  var pledgeAmountViewHidden: Signal<Bool, Never> { get }
+  var shippingLocationViewHidden: Signal<Bool, Never> { get }
+  var shippingSummaryViewHidden: Signal<Bool, Never> { get }
+}
+
+public protocol ConfirmDetailsViewModelType {
+  var inputs: ConfirmDetailsViewModelInputs { get }
+  var outputs: ConfirmDetailsViewModelOutputs { get }
+}
+
+public class ConfirmDetailsViewModel: ConfirmDetailsViewModelType, ConfirmDetailsViewModelInputs,
+  ConfirmDetailsViewModelOutputs {
+  public init() {
+    let initialData = Signal.combineLatest(
+      self.configureWithDataProperty.signal,
+      self.viewDidLoadProperty.signal
+    )
+    .map(first)
```

Comment:
Nit: This is a pattern we use everywhere, but every time I see us using `first`, I assume this is a static/unchanging value, and should really just be passed in on init for the view model. However, again, I assume this is more expedient to write since you're following established patterns, so this is non-blocking.